kubernetes/deployment/controller_test.go

```diff
@@ -0,0 +1,172 @@
+	}{
+		{
+			containerImage:  "ubuntu:latest",
+			tempMountPath:   "/wr-tmp/",
+			cmdArgs:         []string{"/wr-tmp/wr", "manager", "start", "-f"},
+			configMountPath: "/scripts/",
+			configMapData:   "echo \"hello world\"",
+			requiredPorts:   []int{8080, 8081},
+			resourcePath:    "/tmp/resources",
+		},
+	}
+	for _, c := range cases {
+
+		// Test the creation of config maps (2 birds one stone). We won't delete
+		// this now so we can use it later.
+		configmap, err := dc.Client.CreateInitScriptConfigMap(c.configMapData)
+		if err != nil {
+			t.Error(err.Error())
+		}
+
+		expectedData := "#!/usr/bin/env bash\nset -euo pipefail\necho \"Running init script\"" +
+			"\necho \"hello world\"\necho \"Init Script complete, executing arguments provided\"\nexec $@"
+
+		if configmap.Data[client.DefaultScriptName] != expectedData {
+			t.Error(fmt.Errorf("Unexpected contents of config map, got:\n%s \nexpect:\n%s", configmap.Data[client.DefaultScriptName], expectedData))
+		}
+
+		// Create the deployment we run the init script created from wherever
+		// we've decided to mount it.
+		err = dc.Client.Deploy(c.containerImage, c.tempMountPath,
+			c.configMountPath+client.DefaultScriptName,
+			c.cmdArgs, configmap.ObjectMeta.Name,
+			c.configMountPath, c.requiredPorts)
+		if err != nil {
+			t.Error(err.Error())
+		}
+
+		// Now the deployment will be waiting for an attach to copy the binary
+		// to boot from.
+
+		// Create empty resource file:
+		resources := &cloud.Resources{
+			ResourceName: "Kubernetes",
+			Details:      make(map[string]string),
+			PrivateKey:   "",
+			Servers:      make(map[string]*cloud.Server)}
+
+		file, err := os.OpenFile(c.resourcePath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+		if err != nil {
+			t.Error(fmt.Errorf("failed to open resource file %s for writing: %s", c.resourcePath, err))
+		}
+
+		encoder := gob.NewEncoder(file)
+		err = encoder.Encode(resources)
+		if err != nil {
+			t.Error(fmt.Errorf("Failed to encode resource file: %s", err))
+		}
+
+		// Generate certs
+		caFile := "/tmp/ca.pem"
+		certFile := "/tmp/cert.pem"
+		keyFile := "/tmp/key.pem"
+		wrDir := "/wr-tmp/.wr_production/"
+		err = internal.GenerateCerts(caFile, certFile, keyFile, "localhost")
+		if err != nil {
+			t.Errorf("failed to generate certificates: %s", err)
+		}
+
+		dc.Opts = &kubedeployment.DeployOpts{
+			Files:         []client.FilePair{{"/tmp/wr", "/wr-tmp/"}, {caFile, wrDir}, {certFile, wrDir}, {keyFile, wrDir}},
+			RequiredPorts: c.requiredPorts,
+			ResourcePath:  c.resourcePath,
+			Logger:        log15.New(),
+		}
+
+		// Start Controller
+		stopCh := make(chan struct{})
+		defer close(stopCh)
+		go func() {
+			dc.Run(stopCh)
+		}()
+
+		// Test if the manager is up.
+		jq, err := jobqueue.Connect("localhost:8080", caFile, "localhost", []byte{}, 27*time.Second)
+		if err != nil {
+			t.Errorf("Failed to connect to jobqueue: %s", err)
+		}
+
+		t.Logf("jobqueue server mode: %s", jq.ServerInfo.Mode)
```

Comment:
Rather than just log the Mode, perhaps make it a test?

Reply:
I can't make a stand alone test, so its just another condition for the deploy test.